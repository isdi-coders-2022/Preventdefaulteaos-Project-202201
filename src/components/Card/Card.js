import Button from "../Button/Button";
import PropTypes from "prop-types";
import CardComponent from "../../jsStyles/CardStyles";
import useMagicApi from "../../hooks/useMagicApi";
import { useState } from "react";

const Card = ({ card: { imageUrl, name, types, colors, rarity } }) => {
  const { addCardsAPI } = useMagicApi();
  const [isAdded, setIsAdded] = useState("");

  const addCards = () => {
    setIsAdded("added");
    addCardsAPI({ imageUrl, name, types, colors, rarity });
  };

  return (
    <CardComponent className="card">
      <img src={imageUrl} alt={`${name} card`}></img>
      <section>
        <h3 className="card--name">{name}</h3>
        <p className="card--type">Type: {types}</p>
        <p className="card--color">Color: {colors}</p>
        <p className="card--rarity">Rarity: {rarity}</p>
        <Button type="onCard" text="MORE INFO" />
        <Button
          className={isAdded}
          type="onCard"
          text="ADD TO DECK"
          actionOnClick={addCards}
        />
      </section>
    </CardComponent>
  );
};

Card.propTypes = {
  card: PropTypes.shape({
    imageUrl: PropTypes.string,
    name: PropTypes.string,
    types: PropTypes.array,
    colors: PropTypes.array,
    rarity: PropTypes.string,
  }),
};

export default Card;
