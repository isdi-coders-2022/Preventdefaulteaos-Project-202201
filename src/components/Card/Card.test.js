import { render, screen } from "@testing-library/react";
import Card from "./Card";

describe("Given a Card component", () => {
  const givenCardObject = {
    name: "Overtaker",
    manaCost: "{1}{U}",
    cmc: 2.0,
    colors: ["Blue"],
    colorIdentity: ["U"],
    type: "Creature â€” Merfolk Spellshaper",
    types: ["Creature"],
    subtypes: ["Merfolk", "Spellshaper"],
    rarity: "Rare",
    set: "PMMQ",
    setName: "Mercadian Masques Promos",
    text: "{3}{U}, {T}, Discard a card: Untap target creature and gain control of it until end of turn. That creature gains haste until end of turn.",
    flavor: "Its no challenge to master a Mercadian.",
    artist: "Clyde Caldwell",
    number: "89",
    power: "1",
    toughness: "1",
    layout: "normal",
    multiverseid: "97051",
    imageUrl:
      "https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=97051&type=card",
    printings: ["MMQ", "PMMQ", "PRM"],
    originalText:
      "o3oU, ocT, Discard a card from your hand: Untap target creature and gain control of it until end of turn. That creature gains haste until end of turn. (It may attack and ocT the turn it comes under your control.)",
    originalType: "Creature - Spellshaper",
    legalities: [
      {
        format: "Commander",
        legality: "Legal",
      },
      {
        format: "Duel",
        legality: "Legal",
      },
      {
        format: "Legacy",
        legality: "Legal",
      },
      {
        format: "Premodern",
        legality: "Legal",
      },
      {
        format: "Vintage",
        legality: "Legal",
      },
    ],
    id: "5ad15555-fd6f-507a-9b90-92e9ad23d014",
  };
  describe("When it's instanciated with a givenCardObject", () => {
    test("Then it should render the card image", () => {
      render(<Card card={givenCardObject} />);

      const buttonRender = screen.getByRole("img");

      expect(buttonRender).toBeInTheDocument();
    });

    test("Then it should render the card with the name 'Overtaker'", () => {
      const cardName = "Overtaker";

      render(<Card card={givenCardObject} />);

      const buttonRender = screen.getByRole("heading", { name: cardName });

      expect(buttonRender).toBeInTheDocument();
    });
  });
});
