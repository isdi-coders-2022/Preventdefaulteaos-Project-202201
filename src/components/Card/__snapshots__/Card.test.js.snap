// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Card component When executing the component Then it should render 1`] = `
<div
  className="sc-jRQBWg eTyaUi card"
>
  <img
    alt="Overtaker card"
    src="https://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=97051&type=card"
  />
  <section>
    <h3
      className="card--name"
    >
      Overtaker
    </h3>
    <p
      className="card--type"
    >
      Type: 
      Creature
    </p>
    <p
      className="card--color"
    >
      Color: 
      Blue
    </p>
    <p
      className="card--rarity"
    >
      Rarity: 
      Rare
    </p>
    <button
      className="sc-bdvvtL sc-dkPtRN ewqzMe jGLYfy"
    >
      MORE INFO
    </button>
    <button
      className="sc-bdvvtL sc-dkPtRN ewqzMe jGLYfy"
    >
      ADD TO DECK
    </button>
  </section>
</div>
`;
